### DISCUSSION TITLE
What's purpose of grammars and queries under runtime path?

### DISCUSSION THREAD

**Discussion by zjp-CN** [helix-editor/helix#4345] [Q&A] [answered]
Just curious about it. I'm not familiar with tree-sitter at all.
Helix seems to work with no grammars?

**Comment by the-mikedavis** [comment, marked as answer]
The objects under `runtime/grammars` are compiled C/C++ tree-sitter parsers. Helix uses them to parse documents and then uses the tree-sitter queries under `runtime/queries` to query the parsed syntax trees to perform syntax highlighting or determine textobjects or custom auto-indent behavior.

These are in the runtime directory and not compiled directly into Helix because they would bloat the Helix binary a lot and be immutable - you would need to recompile Helix in order to add new language support.

Helix can work without these but all languages effectively become plaintext (other than LSP features).
